drivers/rf-transceiver/hmc630x/iio_hmc630x.c

```diff
@@ -0,0 +1,378 @@
+#include <stdlib.h>
+#include <stdio.h>
+#include <inttypes.h>
+#include <math.h>
+#include "no_os_error.h"
+#include "no_os_delay.h"
+#include "iio.h"
+#include "iio_hmc630x.h"
+#include "no_os_util.h"
+#include "hmc630x.h"
+#include <string.h>
+
+static int32_t _hmc630x_read_register2(struct hmc630x_iio_dev *iiodev,
+				       uint32_t reg,
+				       uint32_t *readval)
+{
+	int ret;
+	uint8_t val;
+	ret = hmc630x_read_row(iiodev->dev, reg, &val);
+	if (ret)
+		return ret;
+
+	*readval = val;
+	return 0;
+}
+
+static int32_t _hmc630x_write_register2(struct hmc630x_iio_dev *iiodev,
+					uint32_t reg,
+					uint32_t writeval)
+{
+	uint8_t val = writeval;
+	return hmc630x_write_row(iiodev->dev, reg, val);
+}
+
+static int hmc630x_iio_read_attr(void *device, char *buf,
+				 uint32_t len, const struct iio_ch_info *channel,
+				 intptr_t priv)
+{
+	int ret;
+	struct hmc630x_iio_dev *iiodev = (struct hmc630x_iio_dev *)device;
+	struct hmc630x_dev *d = (struct hmc630x_dev *)iiodev->dev;
+	int32_t val;
+
+	switch (priv) {
+	case HMC630X_IIO_ATTR_ENABLED:
+		bool enable;
```

Comment:
This is a compilation error (same for all the other variable declarations in this `switch`).